imap_processing/spice/pointing_frame_handler.py

```diff
@@ -0,0 +1,245 @@
+        # but it's not straight forward.  We'll revisit this if needed.
+
+        # Rotation matrix from IMAP spacecraft frame to ECLIPJ2000.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.pxform
+        body_rots = spice.pxform("IMAP_SPACECRAFT", "ECLIPJ2000", tdb)
+        # Convert rotation matrix to quaternion.
+        # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.m2q
+        body_quat = spice.m2q(body_rots)
+        # z-axis of the ECLIPJ2000 frame.
+        z_eclip_time.append(body_rots[:, 2])
+
+        # Standardize the quaternion so that they may be compared.
+        body_quat = body_quat * np.sign(body_quat[0])
+
+        # Aggregate quaternions into a single matrix.
+        aggregate += np.outer(body_quat, body_quat)
+
+    # Reference: "On Averaging Rotations"
+    # Link: https://link.springer.com/content/pdf/10.1023/A:1011129215388.pdf
+    aggregate /= len(et_times)
+
+    # Compute eigen values and vectors of the matrix A
+    # Eigenvalues tell you how much "influence" each
+    # direction (eigenvector) has.
+    # The largest eigenvalue corresponds to the direction
+    # that has the most influence.
+    # The eigenvector corresponding to the largest
+    # eigenvalue points in the direction that has the most
+    # combined rotation influence.
+    eigvals, eigvecs = np.linalg.eig(aggregate)
+    # q0: The scalar part of the quaternion.
+    # q1, q2, q3: The vector part of the quaternion.
+    q_avg = eigvecs[:, np.argmax(eigvals)]
+
+    return q_avg, z_eclip_time
+
+
+def create_rotation_matrix(et_times: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Create a rotation matrix.
+
+    Parameters
+    ----------
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+
+    Returns
+    -------
+    rotation_matrix : np.ndarray
+        Rotation matrix.
+    z_avg : np.ndarray
+        Inertial z axis. Used for plotting.
+    """
+    # Averaged quaternions.
+    q_avg, _ = average_quaternions(et_times)
+
+    # Converts the averaged quaternion (q_avg) into a rotation matrix
+    # and get inertial z axis.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.q2m
+    z_avg = spice.q2m(list(q_avg))[:, 2]
+    # y_avg is perpendicular to both z_avg and the standard Z-axis.
+    y_avg = np.cross(z_avg, [0, 0, 1])
+    # x_avg is perpendicular to y_avg and z_avg.
+    x_avg = np.cross(y_avg, z_avg)
+
+    # Construct the rotation matrix from x_avg, y_avg, z_avg
+    rotation_matrix = np.array([x_avg, y_avg, z_avg])
+
+    return rotation_matrix, z_avg
+
+
+def create_pointing_frame() -> Path:
+    """
+    Create the pointing frame.
+
+    Returns
+    -------
+    path_to_pointing_frame : Path
+        Path to dps frame.
+    """
+    # Mount path to EFS.
+    mount_path = Path(os.getenv("EFS_MOUNT_PATH", ""))
```

Comment:
This should be obtained from `imap_data_access`, or I think it is actually called `IMAP_SPICE_DIR` I think? EFS is an implementation detail, if I have it locally I don't have an EFS MOUNT PATH, I have a SPICE DIR. This is where the `ensure_spice` decorator should point to this location and not require you to manage it here I think.

Reply:
I will put it as a TODO to use ensure_spice.

Reply:
Why not use it now? It is a helper function that should make our lives easier and to me this is currently not correct because it is using an undocumented environment variable. If the decorator isn't helping we should add to it so it is more helpful. @subagonsouth do you have any pointers for how to explicitly use it here?

Reply:
Sure...
```
@ensure_spice()
def create_pointing_frame() -> Path:
```
Then just remove all of the kernel and furnishing code from this function. For now, you will need to furnish the needed kernels in your test functions before calling. I am going to write up a ticket for making sure that we correctly handle furnishing kernels for this specific case in the decorator.